Calendar/BookingHelpers/ExcelWrite.ps1

```diff
@@ -0,0 +1,123 @@
+
+    Write-Host "Exporting Working Hours"
+    $WorkingHours = GetGraphBookingBusinessWorkingHours -Identity $identity
+    $WorkingHours | Export-Excel -Path  $path -WorksheetName "Business" -BoldTopRow  -StartRow 25  -Title "Working Hours" -TableStyle Medium24 -AutoSize
+
+    Write-Host "Exporting Staff"
+    $Staff = Get-GraphBookingsStaff -Identity $identity
+    $Staff | Export-Excel -Path  $path -WorksheetName "Staff" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -Title "Staff List"
+
+    Write-Host "Exporting Services"
+    $Services = Get-GraphBookingsServices -Identity $identity
+    $Services | Export-Excel -Path  $path -WorksheetName "Services" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -Title "Services"
+
+    Write-Host "Exporting Custom Questions"
+    #Get-MgBookingBusinessCustomQuestion -BookingBusinessId $identity | Export-Excel -Path  $path -WorksheetName "Custom Questions" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd
+    $CustomQuestions = Get-GraphBookingsCustomQuestions -Identity $identity
+    $CustomQuestions | Export-Excel -Path  $path -WorksheetName "Custom Questions" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -Title "Custom Questions"
+
+    Write-Host "Exporting MessageTrace for the past $($script:MessageTrackingDays) days"
+    $script:MessageTrackingLogs | Export-Excel -Path  $path -WorksheetName "Message Traces" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3
+
+    Write-Host "Exporting Membership Log data to Excel"
+    $ArrayFromStaffLog = createArrayFromStaffLog -log $script:BookingStaffMembershipLog.MailboxLog
+    $ArrayFromStaffLog | Export-Excel -Path  $path -WorksheetName "Membership Log" -TableStyle Medium3 -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize  -Title "Staff Membership Log"
+
+    Write-Host "Exporting Internal data to Excel"
+    $script:TenantSettings   | Export-Excel -Path  $path -WorksheetName "Tenant Settings - Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "Tenant Settings - Internal"
+    $script:bookingMBData  | Export-Excel -Path  $path -WorksheetName "Mailbox Data - Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "Mailbox Data - Internal"
+    $script:MBPermissions | Export-Excel -Path  $path -WorksheetName "MB Permissions-Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-Title "MB Permissions - Internal"
+    $script:MBRecipientPermissions | Export-Excel -Path  $path -WorksheetName "Recipient Permissions-Internal" -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd -AutoSize -TableStyle Medium3 #-tableTitle "Recipient Permissions - Internal"
+
+    # $dateNow = Get-Date
```

Comment:
Is this sample code or should it be removed?

Reply:
Leaving for now for initial tests with support SME's. Possibly moving to Logic Functions in the future.
Pending PG modification to graph availability to allow to get availability as delegated permissions and not having to use application permissions.